### PR TITLE
ImGui - Menus

### PR DESCRIPTION
**Pull request by supudo**
Hey,

User of ImGui here. Was looking for a Python port and yours seems to truly work :)
I noticed there's still plenty of porting to be done, so as my first contribution ever i'm pushing the Menus functions ported.

If you could please check and see if i'm not doing some ugly work (Python is not my strong language) and maybe to merge the changes :)

Thanks and waiting for your feedback. Again - amazing work with porting ImGui!

Sergey

### CODE REVIEW

**Comment by swistakm** [comment]
Thanks for the contribution! This will bump the completion of the library by another 2%.

I will do review since I have already noticed few issues. Mostly there are issues with documentation and examples defined with the `.. visual-example::` directive.

There are no guidelines for writing them yet but they are important because serve as a kind of tests for the library (unit-testing of GUI libraries is always tricky). But don't worry - I will guide you through the whole feature and maybe this will result with some documentation for future contributors.

**Comment by swistakm** [comment]
Note that you can build sphinx documentation to see if visual examples compile using:

    make build && make -C doc/ html

It can be viewed in your web browser from `doc/build/html/index.html`. Ignore resulting png files and don't add them to the PR.

**Comment by supudo** [comment]
Thanks for the comment!

I was trying to follow the docs, but it's obvious i'm not familiar with it :).
Will check your changes and will follow them in future commits.

S.

**Comment by swistakm** [comment]
Review done. I hope you won't be overwhelmed by the number of my comments :)

**Comment by swistakm** [comment]
Additional note: I have just merged #5 that was waiting very long for review. It ensures Python 2/3 compatibility. In order to avoid conflicts and ensure py3 compat you will have to rebase your changes and change the way how string labels/names are handled (see diff in the #5 PR for reference).

File: imgui/core.pyx [hunk 1]
```diff
@@ -1365,6 +1365,196 @@ def is_window_collapsed():
...
+    Returns:
+
+    .. wraps::
+         void EndMenu()
+    )
+    """
+    cimgui.EndMenu()
+
+
+def menu_item(char* name, char* shortcut='', selected=False, enabled=True):
```
Comment (swistakm) [inline imgui/core.pyx]: Note that this is the design problem that bothered me for a while and is also reason why my development of this project slowed down. I have created a design document (see: #8) to organise my thoughts on this issue. You can comment and add some suggestions/opinions if you are interested.

**Pull request closed without merging**